src/openrct2/object/SceneryGroupObject.cpp

```diff
@@ -192,7 +192,35 @@ std::vector<ObjectEntryDescriptor> SceneryGroupObject::ReadJsonEntries(json_t& j
 
     for (const auto& jEntry : jEntries)
     {
-        entries.emplace_back(Json::GetString(jEntry));
+        auto entryName = Json::GetString(jEntry);
+        if (String::StartsWith(entryName, "$DAT:"))
+        {
+            if (entryName.length() != 5 + 8 + 1 + 8)
+            {
+                log_error("Malformed DAT entry in scenery group: %s", entryName.c_str());
+                continue;
+            }
+
+            try
+            {
+                auto originalName = entryName.substr(5 + 8 + 1, 8);
+
+                rct_object_entry entry = {};
+                entry.flags = std::stoul(entryName.substr(5, 8), nullptr, 16);
+                entry.checksum = 0;
+                auto minLength = std::min<size_t>(8, originalName.length());
```

Comment:
Since you only allow strings that match the exact length you expect and your substring above is exactly 8 characters, I can't see how this would be anything else.

But more importantly, it is inefficient to create a new `std::string` for originalName since you `memcpy` it anyway into the object entry. Why don't you just memcpy directly from `entryName`?

Reply:
I’ll try doing that.

It is like this because I directly copied this code from `ObjectFactory.cpp`.